pinecone/client.go

```diff
@@ -132,6 +400,62 @@ func (c *Client) ListIndexes(ctx context.Context) ([]*Index, error) {
 	return indexes, nil
 }
 
+// CreatePodIndexRequest holds the parameters for creating a new pods-based index.
+//
+// Fields:
+//   - Name: The name of the pods-based index to be created.
+//   - Dimension: The dimension of the index (must match [dimensionality] of upserted vectors).
+//   - Metric: The metric used to measure the [similarity] between vectors.
+//   - Environment: The [cloud environment] in which the index will be created.
+//   - PodType: The [type of pod] to use for the index.
```

Comment:
I'd maybe lift the specific verbiage from the OAS spec: 

`The type of pod to use. One of `s1`, `p1`, or `p2` appended with `.` and one of `x1`, `x2`, `x4`, or `x8`.